src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -1337,12 +1337,18 @@ public void setAuthenticator(Authenticator authenticator) {
 
     @InterfaceAudience.Private
     protected boolean serverIsSyncGatewayVersion(String minVersion) {
+        return serverIsSyncGatewayVersion(serverType, minVersion);
+    }
+    
+    @InterfaceAudience.Private
+    protected static boolean serverIsSyncGatewayVersion(String serverName, String minVersion) {
         String prefix = "Couchbase Sync Gateway/";
-        if (serverType == null) {
+        if (serverName == null) {
             return false;
         } else {
-            if (serverType.startsWith(prefix)) {
-                String versionString = serverType.substring(prefix.length());
+            if (serverName.startsWith(prefix)) {
+                String versionString = serverName.substring(prefix.length());
+                // NOTE: If version number is higher than 10.xx, this comprison does not work eg. "10.0" < "2.0"
```

Comment:
This is a future bug, but it will be very long time to reach v.10 so probably OK.